plugin/trino-resource-group-managers/src/main/java/io/trino/plugin/resourcegroups/StaticSelector.java

```diff
@@ -86,6 +88,7 @@ public StaticSelector(
                 }))
                 .add(userGroupRegex.map(userGroupRegexValue ->
                             new BasicMatcher(criteria -> criteria.getUserGroups().stream().anyMatch(userGroup -> userGroupRegexValue.matcher(userGroup).matches()))))
+                .add(queryTextRegex.map(queryTextRegexValue -> new PatternMatcher(variableNames, queryTextRegexValue, SelectionCriteria::getQueryText)))
```

Comment:
Should we call `addNamedGroups` on the pattern, like we do with `sourceRegex` etc.?